felt_python/layers.py

```diff
@@ -57,50 +63,69 @@ def _multipart_request(
     return urllib.request.Request(url, data=body, headers=headers, method="POST")
 
 
-def _request_and_upload(
```

Comment:
I think I prefer having this utility function instead of repeating the presigned-post-multipart-request logic in both `upload_file` and `refresh_file_layer` but happy to be convinced otherwise!